ql/src/test/queries/clientpositive/cbo_case_when_wrong_type.q

```diff
@@ -0,0 +1,20 @@
+create external table test_case (row_seq smallint, row_desc string) stored as parquet;
```

Comment:
can we also add a simpler testcase like:
```
explain select 1 from t where a in (1,2,3) and case a when 1S then true when 2S then true end;
```

Reply:
Sure, would you prefer a separate qtest file or to integrate it into this one?